config-ui/src/hooks/data-scope/useTransformationsManager.jsx

```diff
@@ -0,0 +1,237 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { useCallback, useState } from 'react'
+import { Providers } from '@/data/Providers'
+import useDataScopesManager from '@/hooks/useDataScopesManager'
+import TransformationSettings from '@/models/TransformationSettings'
+import { isEqual } from 'lodash'
+
+// TODO separate to each plugin
+const getDefaultTransformations = (provider) => {
+  let transforms = {}
+  switch (provider) {
+    case Providers.GITHUB:
+      transforms = {
+        prType: '',
+        prComponent: '',
+        prBodyClosePattern: '',
+        issueSeverity: '',
+        issueComponent: '',
+        issuePriority: '',
+        issueTypeRequirement: '',
+        issueTypeBug: '',
+        issueTypeIncident: '',
+        refdiff: null,
+        productionPattern: '',
+        deploymentPattern: ''
+        // stagingPattern: '',
+        // testingPattern: ''
+      }
+      break
+    case Providers.JIRA:
+      transforms = {
+        epicKeyField: '',
+        typeMappings: {},
+        storyPointField: '',
+        remotelinkCommitShaPattern: '',
+        bugTags: [],
+        incidentTags: [],
+        requirementTags: [],
+        // @todo: verify if jira utilizes deploy tag(s)?
+        productionPattern: '',
+        deploymentPattern: ''
+        // stagingPattern: '',
+        // testingPattern: ''
+      }
+      break
+    case Providers.JENKINS:
+      transforms = {
+        productionPattern: '',
+        deploymentPattern: ''
+        // stagingPattern: '',
+        // testingPattern: ''
+      }
+      break
+    case Providers.GITLAB:
+      transforms = {
+        productionPattern: '',
+        deploymentPattern: ''
+        // stagingPattern: '',
+        // testingPattern: ''
+      }
+      break
+    case Providers.TAPD:
+      // @todo: complete tapd transforms #2673
+      transforms = {
+        issueTypeRequirement: '',
+        issueTypeBug: '',
+        issueTypeIncident: '',
+        productionPattern: '',
+        deploymentPattern: ''
+        // stagingPattern: '',
+        // testingPattern: ''
+      }
+      break
+  }
+  return transforms
+}
+
+// manage transformations in one place
+const useTransformationsManager = () => {
+  const [transformations, setTransformations] = useState({})
+
+  const generateKey = (
+    connectionProvider,
+    connectionId,
+    projectNameOrBoard
+  ) => {
+    let key = `not-distinguish`
```

Comment:
What benefit do we have of having initial key default to `not-distinguish`? Why not default to Connection ID like I used previously? `C#[connId]` as a sane value to be used for providers like Jenkins which started with no project or entity concept?
```suggestion
    let key = `${connectionProvider}/${connectionId}`
```

Reply:
Now key will be `${connectionProvider}/${connectionId}/not-distinguish`. That will be more readable when the new developer adds another plugin next time but forgets to distinguish `connectionId`. He will find here move easily.

Reply:
Ok, as a grammatical suggestion `not-distinguished` in past tense may read a bit better.